### PR TITLE
Update_to_robotpy_2025

### PR DESCRIPTION
**Pull request by LeanMeanBeanMachine4774**

### CODE REVIEW

**Comment by LeanMeanBeanMachine4774** [comment]
UV sync worked when allowing prereleases for dependencies

**Comment by james-ward** [comment]
LGTM, but I'll let @LucienMorey have the last say. I don't want to mess up his demo on Sunday (I don't think it will).

File: physics.py [hunk 1]
```diff
@@ -44,11 +44,12 @@ def __init__(
         gearing: float,
         moi: kilogram_square_meters,
     ):
+        self.plant = LinearSystemId.DCMotorSystem(DCMotor.falcon500(), moi, gearing)
```
Comment (LucienMorey) [inline physics.py:47]: I think you may want to create a single motor and pass it here and to the sim. Things might get a bit funny if those things somehow diverge in state

**Review by LucienMorey — APPROVED** [review]

File: pyproject.toml [hunk 2]
```diff
@@ -73,25 +74,25 @@ license = {text = "MIT"}
 requires-python = ">=3.10,<3.13"
 
 dependencies = [
-    "numpy~=1.25",
-    "phoenix6~=24.2.0",
+    "numpy~=2.1",
+    "phoenix6~=25.0.0b2",
     # robotpy[apriltag] pins numpy to a version that doesn't build cleanly on Python 3.12.
```
Comment (auscompgeek) [inline pyproject.toml:79]: This comment is no longer true.

- https://github.com/robotpy/robotpy-meta/blob/6aa03b1f4800f070c1c05ad3d9eddf6b4754ad35/pyproject.toml#L140-L141
- https://pypi.org/project/numpy/2.1.2/#files

```suggestion
```

But maybe you should consider using `robotpy[apriltag]` in this list now that's fixed.

**Pull request merged**